core/src/epicli/data/common/ansible/playbooks/roles/repository/tasks/setup.yml

```
@@ -76,3 +76,7 @@
 
 - name: Set up repositories
   include_tasks: "{{ ansible_os_family }}/setup.yml"
+
+- name: Include Helm repository creation
+  include_tasks: "create-helm-repo.yml"
+  when: inventory_hostname == target_repository_hostnames[0]
```

```suggestion
  when: inventory_hostname in target_repository_hostnames
```
How about keeping the same condition for consistency?
```
- name: Produce final list of hosts to deploy repositories on
  set_fact:
    target_repository_hostnames: >-
      {{ groups.repository | difference(groups.kubernetes_master) | union([resolved_repository_hostname]) | unique }}
```